channeldb/payments.go

```diff
@@ -525,10 +528,18 @@ func serializeHop(w io.Writer, h *route.Hop) error {
 		return WriteElements(w, uint32(0))
 	}
 
+	// Gather all non-primitive TLV records so that they can be serialized
+	// as a single blob.
+	var records []tlv.Record
+	if h.MPP != nil {
+		records = append(records, h.MPP.Record())
+	}
+	records = append(records, h.TLVRecords...)
```

Comment:
`TLVRecords` is for custom fields offered via rpc. Since this is a known field we extract it. Unfortunately we will have to do this for every known field going forward...